src/util/strencodings.h

```diff
@@ -94,6 +96,24 @@ constexpr inline bool IsSpace(char c) noexcept {
     return c == ' ' || c == '\f' || c == '\n' || c == '\r' || c == '\t' || c == '\v';
 }
 
+/**
+ * Convert string to integral type T.
+ *
+ * @returns std::nullopt if the entire string could not be parsed, or if the
+ *   parsed value is not in the range representable by the type T.
+ */
+template <typename T>
+std::optional<T> ToIntegral(const std::string& str)
```

Comment:
I am wondering if it makes sense to explain in simple terms the format that is required to be parsed successfully. In regex terms it would be `-?[0-9]+`?